string manipulator
---
RobotStateBase state_with_base
